### DISCUSSION TITLE
Cannot read properties of undefined

### DISCUSSION THREAD

**Discussion by aubriewynbae** [requarks/wiki#4677] [Error / Bug Report]
**Describe the bug**
Nonspecific error is preventing import from local repo.

**To Reproduce**
I'm not really sure how it got to this point, as I've tested two deploys thus far and only one has this issue. I've installed Wiki.js with its own user, it connects to the database just fine, it connects to the Git repo just fine, it pulls the repo just fine locally, but as soon as I try to run an import, the import starts, then stalls immediately after a **warn: Cannot read properties of undefined (reading 'split')** error. I'm not sure where to continue troubleshooting as that's the only error that comes up. Is there any other log file that would be worth checking?

**Expected behavior**
Expecting the import to complete normally as it did on the first deploy (identical to this one).

**Logs**
```
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.765Z [MASTER] info: (STORAGE/GIT) Importing all content from local Git repo to the DB...
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.768Z [MASTER] info: (STORAGE/GIT) Processing Campaigns...
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.769Z [MASTER] info: (STORAGE/GIT) Asset marked as renamed: from undefined to Campaigns
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.771Z [MASTER] info: (STORAGE/GIT) Asset was not found in the DB, nothing to rename: Campaigns
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.771Z [MASTER] info: (STORAGE/GIT) Processing Classes...
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.772Z [MASTER] info: (STORAGE/GIT) Asset marked as renamed: from undefined to Classes
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.773Z [MASTER] info: (STORAGE/GIT) Asset was not found in the DB, nothing to rename: Classes
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.773Z [MASTER] info: (STORAGE/GIT) Processing Classes.md...
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.776Z [MASTER] info: (STORAGE/GIT) Page marked as renamed: from undefined to Classes.md
Nov 09 04:34:45 localhost node[607]: 2021-11-09T04:34:45.777Z [MASTER] warn: Cannot read properties of undefined (reading 'split')
```

**Host Info (please complete the following information):**
 - OS: Ubuntu 20.04.1
 - Wiki.js version: 2.5.219
 - Database engine: Postgres 12.8

**Comment by nuraeil** [comment]
I've also been experiencing this issue now on two different deployments. 

It also is not removing deleted files from the database that were delete inside the repo folder... When trying to delete those pages in the system directly, it says they already do not exist which is correct, but the page is still being displayed and listed.

![image](https://user-images.githubusercontent.com/32387849/140875598-d61e33eb-6d00-4abb-96fb-51770b817f19.png)

And at the current scope, I would have to manually delete 100+ old pages to have them removed fully from the system.

It simply stops syncing any changes and throws this error, not continuing with the sync process. It partly imports some files... But I changed around 100 files and maybe ~5 of them were synced into the database.

I have tried fully rebasing it, renaming files, etc. I also cleared the cache and tried rebuilding the sites, but it does not even recognize that these pages were deleted in the sync from the Git repo...

Things I have tried so far:

* Purged the local Repository
* Tried importing, simply stops and throws the warning and cannot read properties of undefined
* Can not even dump the database into the local, it also throws the same error
* Rebuild the Page Tree
* Rerender All Pages
* Purge Page History
* Flushed Pages and Assets Cache
* Flush Temporary Uploads
* Flush Client-Side Locale Cache
* Setup Git Storage Sync on a fresh deployment, fails with the same error

This currently breaks our workflow and we were in the process of going productive with the system, which is now on ice till this issue is fixed.

It was working before... I believe something in the last 3 or 4 releases somehow broke the Storage sync.

And logging is defaulted to info, so only way for me to get more information would be mapping a config map for the config.yml to override logging.

@aubriewynbae: You can change the log-level inside the config.yml file from info to debug. Otherwise you can only log the queries to the database and the ldap, which shows nothing as I tried that. See https://docs.requarks.io/en/install/config#logging

Host Info (please complete the following information):

OS: Docker Container (Linux)
Wiki.js version: 2.5.219
Database engine: postgresql-repmgr:11.13.0-debian-10-r57 (Bitnami Postgresql-HA v7.12.4)

**Reply by nuraeil** [reply]
Following up on this: I just basically deleted every single page manually, fully cleared the git repo and then pushed everything again into it... Now it synced everything... This took me about 3 hours as we had around 400 pages...

**Comment by EricFromCanada** [comment]
Potentially fixed in [v2.5.296](https://github.com/requarks/wiki/releases/tag/v2.5.296) via https://github.com/requarks/wiki/pull/6020
